### PR TITLE
Add a Package.swift for SwiftPM support

### PR DESCRIPTION
**Pull request by j-f1**
This allows llama.cpp to be called directly from Swift! First add `https://github.com/ggerganov/llama.cpp` to your `Package.swift` or Xcode project, selecting either this branch or `master` (once the PR is merged).

Here’s a basic example of calling from Swift:

```swift
import llama

let ctx = llama_init_from_file(url.path(percentEncoded: false), llama_context_default_params())

let promptTokens = Array<llama_token>(unsafeUninitializedCapacity: prompt.utf8.count) { buffer, initializedCount in
  initializedCount = Int(llama_tokenize(ctx, prompt, buffer.baseAddress, Int32(buffer.count), true))
}
for var token in promptTokens {
  llama_eval(ctx, &token, 1, Int32(tokens.count), nThreads)
}

while true { // should stop after reaching context limit!
  var token = llama_sample_top_p_top_k(ctx, nil, 0, topK, topP, temperature, 1)
  if token == llama_token_eos() {
    print("[end of text]")
    break
  }
  print(String(cString: llama_token_to_str(ctx, token)), terminator: "")
  llama_eval(ctx, &token, 1, Int32(tokens.count), nThreads)
}

llama_free(ctx)
```

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    name: "llama",
+    products: [
+        .library(name: "llama", targets: ["llama"]),
+    ],
+    targets: [
+        .target(
+            name: "llama",
+            path: ".",
+            exclude: ["main.cpp", "tests"],
+            cSettings: [.unsafeFlags(["-Wno-shorten-64-to-32"])]
```
[the remark this answers is not in the document]
Comment (ggerganov) [reply Package.swift:16]: I don't think it is something to worry about

**Comment by j-f1** [comment]
Tested and it works locally!

**Comment by j-f1** [comment]
Is anyone interested in reviewing this?

**Review by Brianm93 — COMMENTED** [review]
Hey

**Pull request merged**